sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Constants.java

```diff
@@ -247,5 +247,5 @@ public static final class QueryExecutionContext {
         public static final String INCREMENTAL_FEED_HEADER_VALUE = "Incremental feed";
     }
 
-    public static final int QUERYPLAN_CACHE_SIZE = 1000;
+    public static final int QUERYPLAN_CACHE_SIZE = 5000;
```

Comment:
This will increase the memory use of people who make a lot of use of queries right?

Reply:
Yeah, it will, however we measured the size of this and it is not a lot even with 5000 queries. Also, query plan cache only works on parametrized queries only (`SqlQuerySpec` based), so customers shouldn't have more than 5000 distinct queries, and if they are using it, they are messed up at first place. 
In any case, we decided to make this change earlier (even go to 10,000) but didn't do. 

Some more information on this PR description about this - (previously we had LRU Cache) - https://github.com/Azure/azure-sdk-for-java/pull/28620#issue-1223544220